TextBlob is a Python library for processing textual data
As TextBlob is a Lexicon-based sentiment analyzer It has some predefined rules or we can say word and weight dictionary, where it has some scores that help to calculate a sentence's polarity. That's why the Lexicon-based sentiment analyzers are also called “Rule-based sentiment analyzers"

VADER is another Lexicon-based sentiment analyzer that has pre-defined rules for words or lexicons. VADER not only tells the lexicon is positive, negative, or neutral, it also tells how positive, negative, or neutral a sentence is.

WordCloud is a pictorial presentation of most frequent words

In [1]:
import pandas as pd
df = pd.read_csv('tripadvisor_hotel_reviews.csv')
df.head()

Review  Rating
0  nice hotel expensive parking got good deal sta...       4
1  ok nothing special charge diamond member hilto...       2
2  nice rooms not 4* experience hotel monaco seat...       3
3  unique, great stay, wonderful time hotel monac...       5
4  great stay great stay, went seahawk game aweso...       5

In [2]:
df.isnull().sum()

Review    0
Rating    0
dtype: int64

In [3]:
df.shape

(20491, 2)

lets say we dont have the 'rating' column only have review col but we want to do sentiment analysis

In [4]:
df.iloc[0][0]   # for first row

'nice hotel expensive parking got good deal stay hotel anniversary, arrived late evening took advice previous reviews did valet parking, check quick easy, little disappointed non-existent view room room clean nice size, bed comfortable woke stiff neck high pillows, not soundproof like heard music room night morning loud bangs doors opening closing hear people talking hallway, maybe just noisy neighbors, aveda bath products nice, did not goldfish stay nice touch taken advantage staying longer, location great walking distance shopping, overall nice experience having pay 40 parking night,  '

We don't need to remove punctuations , stopwords because textblob and vader is a powerfull and intelligent library enough to detach punctuations and stopwords

In [5]:
from textblob import TextBlob
blob = TextBlob(df.iloc[0][0])  # for 0th index

In [6]:
blob.sentiment

Sentiment(polarity=0.20874404761904758, subjectivity=0.687)

In [7]:
blob.sentiment[0]  # its shows polarity

0.20874404761904758

In [8]:
sent_list = []
for i in range(0,df.shape[0]):
    blob = TextBlob(df.iloc[i][0])
    polarity_score = blob.sentiment[0]
    sent_list.append(polarity_score)

In [9]:
df['Sentiment_score'] = sent_list
df.head()  # we added another column in dataframe  last column shows polarity score of each row

Review  Rating  Sentiment_score
0  nice hotel expensive parking got good deal sta...       4         0.208744
1  ok nothing special charge diamond member hilto...       2         0.214923
2  nice rooms not 4* experience hotel monaco seat...       3         0.294420
3  unique, great stay, wonderful time hotel monac...       5         0.504825
4  great stay great stay, went seahawk game aweso...       5         0.384615

In [10]:
df[df['Sentiment_score'] < 0].shape  # count of -ve sentiment 

(1356, 3)

It means there are 1356 reviews are negative

# Vader

The purpose of the vader libary is also same as your textblob , It is a pretrained library used for your sentiment analysis

In [11]:
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer
nltk.download('vader_lexicon')

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     C:\Users\Asus\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


True

In [12]:
# checking
sent = SentimentIntensityAnalyzer()
sent.polarity_scores('cough surup was awesome and i had heavily feeling after having it') 

{'neg': 0.0, 'neu': 0.616, 'pos': 0.384, 'compound': 0.6808}

neg = negative , neu = neutral , pos = positive , compound = overall

In [16]:
sent.polarity_scores(df.iloc[0][0])['compound']

0.9747

In [18]:
compound_list =[] # I want only compound values of all reviews
for i in range(0 , df.shape[0]):
    score = sent.polarity_scores(df.iloc[i][0])
    sent_score = score['compound']
    compound_list.append(sent_score)
len(compound_list)

20491

In [19]:
df['Score_vader'] = compound_list # putting into dataframe

In [20]:
df.head()

Review  Rating  Sentiment_score  \
0  nice hotel expensive parking got good deal sta...       4         0.208744   
1  ok nothing special charge diamond member hilto...       2         0.214923   
2  nice rooms not 4* experience hotel monaco seat...       3         0.294420   
3  unique, great stay, wonderful time hotel monac...       5         0.504825   
4  great stay great stay, went seahawk game aweso...       5         0.384615   

   Score_vader  
0       0.9747  
1       0.9787  
2       0.9889  
3       0.9912  
4       0.9797